metricbeat/helper/prometheus/prometheus_test.go

```diff
@@ -98,6 +98,22 @@ metrics_one_midichlorians_bucket{rank="padawan",alive="yes",le="+Inf"} 28
 metrics_one_midichlorians_sum{rank="padawan",alive="yes"} 800001
 metrics_one_midichlorians_count{rank="padawan",alive="yes"} 28
 
+`
+
+	promHistogramKeyLabelWithNaNInf = `
+# TYPE metrics_one_midichlorians histogram
+metrics_one_midichlorians_bucket{rank="youngling",alive="yes",le="100"} NaN
+metrics_one_midichlorians_bucket{rank="youngling",alive="yes",le="500"} +Inf
+metrics_one_midichlorians_bucket{rank="youngling",alive="yes",le="1000"} -Inf
```

Comment:
It would be really really weird to find these values here, because they are integers, and integers cannot have these values in Go. Also this is a histogram, these values here would be incorrect. In this case this could be considered as invalid as finding an string.

The issue observed in #10849 was with gauges, something like:
```
# HELP kafka_consumer_records_lag_records The latest lag of the partition
# TYPE kafka_consumer_records_lag_records gauge
kafka_consumer_records_lag_records{client_id="some-consumer",} NaN
kafka_consumer_records_lag_records{client_id="other-consumer",} NaN
kafka_consumer_records_lag_records{client_id="another-consumer",} NaN
```

Reply:
Yeah I agree, this should not happen on histogram. I will remove the code as well as the test!